File: lib/Models/mtg/card/mtg_card.dart
import 'dart:convert';
import 'dart:core';

import 'package:tcg_scanner2/Models/mtg/card/mtg_card_foreign_name.dart';
import 'package:tcg_scanner2/Models/mtg/card/mtg_card_legality.dart';
import 'package:tcg_scanner2/Models/mtg/card/mtg_card_ruling.dart';

class MtgCard {
  MtgCard();
   String? id;
   String name = "";
   List<String>? names;
   String manaCost = "";
   int? cmc;
   List<String>? colors;
   List<String>? colorIdentity;
   String type = "";
   List<String>? supertypes;
   List<String> types = [];
   List<String>? subtypes;
   String rarity = "";
   String set = "";
   String setName = "";
   String? text;
   String artist = "";
   String? number;
   String? power;
   String? toughness;
   String? loyalty;
   int? multiverseid;
   String? imageUrl;
   String layout = "";
   List<MtgCardLegality>? legalities;
   List<MtgCardRuling>? rulings;
   List<MtgCardForeignName>? foreignNames;

   factory MtgCard.fromJson(Map<String, dynamic> json) {
     final card = MtgCard();
     card.id = json['id'];
     card.name = json['name'];
     card.names = json['names'];
     card.manaCost = json['manaCost'] ?? "";
     card.cmc = json['cmc'].toInt();
     card.colors = ((json['colors'] ?? []) as List).map((color) => color as String).toList();
     card.colorIdentity = ((json['colorIdentity'] ?? []) as List).map((color) => color.toString()).toList();
     card.type = json['type'];
     // card.supertypes = ((json['supertypes'] ?? [])).map((supertype) => supertype as String).toList();
     card.types = ((json['types'] ?? []) as List).map((type) => type.toString()).toList();
     card.subtypes = ((json['subtypes'] ?? []) as List).map((subtype) => subtype.toString()).toList();
     card.rarity = json['rarity'];
     card.set = json['set'];
     card.setName = json['setName'];
     card.text = json['text'];
     card.artist = json['artist'];
     card.number = json['number'];
     card.power = json['power'];
     card.toughness = json['toughness'];
     card.loyalty = json['loyalty'];
     card.multiverseid = int.parse(json['multiverseid']);
     card.imageUrl = json['imageUrl'];
     card.layout = json['layout'];
     // TODO: Requires testing of each one
     card.legalities = ((json['legalities'] ?? []) as List).map((legal) => MtgCardLegality.fromJson(legal)).toList();
     card.rulings = ((json['rulings'] ?? []) as List).map((rule) => MtgCardRuling.fromJson(rule)).toList();
     card.foreignNames = ((json['foreignNames'] ?? []) as List).map((foreignName) => MtgCardForeignName.fromJson(foreignName)).toList();
     return card;
   }
}

File: TestTcgScanner/TestTcgScanner/Models/MtgCard.cs
using System.Collections.Generic;

namespace TestTcgScanner.Models
{
    public class MtgCard
    {
        public string Name { get; set; }

        public string? Number { get; set; }

        public string Description { get; set; }

        public string? Id { get; set; }
        public List<string>? Names { get; set; }
        public string ManaCost { get; set; } = "";
        public double? Cmc { get; set; }
        public List<string>? Colors { get; set; }
        public List<string>? ColorIdentity { get; set; }
        public string Type { get; set; } = "";
        public List<string>? Supertypes { get; set; }
        public List<string> Types { get; set; }
        public List<string>? Subtypes { get; set; }
        public string Rarity { get; set; } = "";
        public string Set { get; set; } = "";
        public string SetName { get; set; } = "";
        public string? Text { get; set; }
        public string Artist { get; set; } = "";
        public string? Power { get; set; }
        public string? Toughness { get; set; }
        public string? Loyalty { get; set; }
        public int? Multiverseid { get; set; }
        public string? ImageUrl { get; set; }
        public string Layout { get; set; } = "";
        public List<MtgCardLegality>? Legalities { get; set; }
        public List<MtgCardRuling>? Rulings { get; set; }
        public List<MtgCardForeignName>? ForeignNames { get; set; }
    }
}
